tests/kernel/udev_no_symlink.pm

```diff
@@ -0,0 +1,67 @@
+# SUSE's openQA tests
+#
+# Copyright © 2019 SUSE LLC
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved.  This file is offered as-is,
+# without any warranty.
+#
+# Summary: bsc#1089761, SUSE-RU-2018:2620-1
+#
+# Maintainer: Clemens Famulla-Conrad <cfamullaconrad@suse.de>
+
+use Mojo::Base "opensusebasetest";
+use testapi;
+use utils 'zypper_call';
+use bootloader_setup 'add_grub_cmdline_settings';
+use power_action_utils 'power_action';
+
+
+sub run {
+    my ($self) = @_;
+    $self->select_serial_terminal;
+
+    # Normal SLES installation should not have partitions with name logical or primary
+    assert_script_run('test $(grep -r "E:ID_PART_ENTRY_NAME=primary" /run/udev/data | wc -l) -eq 0');
+    assert_script_run('test $(grep -r "E:ID_PART_ENTRY_NAME=logical" /run/udev/data | wc -l) -eq 0');
+    record_info('OK', 'No partion labels found with name equal to primary|logical');
+
+    # Prepare HDD
+    zypper_call('-q in parted');
+    my $hdd2           = '/dev/vdb';
```

Comment:
I know that hardcoding  device name may cause issues because in different systems devices may be named in different way ( vda vs hda vs sda ). Suspecting that this problem may be solved somewhere in our huge codebase.

@okurz , @rwx788 , @Soulofdestiny  any hints ?

Reply:
we've discussed exactly this challenge with @OleksandrOrlov yesterday. script_output can be used with `mount -v | fgrep 'on / '` or `df -k /` for finding disk with root partition. However, I would say it's not causing any issues, as we if we don't change QEMU settings or not starting test on some other backend, we are on the safe side.
Currently I was able to find this one: https://github.com/os-autoinst/os-autoinst-distri-opensuse/blob/master/tests/installation/rescuesystem_validate_sle.pm#L19

Reply:
and what about o3 vs osd ? I know for sure that NICs has different naming ( eth0 vs ens0 ) not sure about HDDs

Reply:
o3 vs osd doesn't make any difference by default, but again, alternatives are mentioned. For NICs it's more complicated, as we have less control over it and those are harder to predict. For HDDs, it's straightforward (unless there are changes to udev rules).

Reply:
ok so let's leave it for now as it is and firefight in case of troubles :)